--!strict
local Option = require("@pkg/@nightcycle/option")
local Result = require("@pkg/@nightcycle/result")
local Future = require("@pkg/@nightcycle/future")
local Error = require("@pkg/@nightcycle/error")
local Vec = require("@pkg/@nightcycle/vec")
local HashMap = require("@pkg/@nightcycle/hash-map")
local VecDeque = require("@pkg/@nightcycle/vec-deque")
local SerdeJson = require("@pkg/@nightcycle/serde-json")
local Reqwest = require("@pkg/@nightcycle/reqwest")

export type Option<T> = Option.Option<T>
export type Result<T, E> = Result.Result<T, E>
export type Future<T> = Future.Future<T>
export type Error<T, Body> = Error.Error<T, Body>
export type Vec<T> = Vec.Vec<T>
export type MutVec<T> = Vec.MutVec<T>
export type HashMap<K, V> = HashMap.HashMap<K, V>
export type MutHashMap<K, V> = HashMap.MutHashMap<K, V>
export type VecDeque<T> = VecDeque.VecDeque<T>
export type Ser = SerdeJson.Ser
export type Deser = SerdeJson.Deser
export type JsonValue = SerdeJson.JsonValue
export type Value = SerdeJson.Value
export type AxesJson = SerdeJson.AxesJson
export type BoolJson = SerdeJson.BoolJson
export type BrickColorJson = SerdeJson.BrickColorJson
export type CFrameJson = SerdeJson.CFrameJson
export type Color3Json = SerdeJson.Color3Json
export type ColorSequenceJson = SerdeJson.ColorSequenceJson
export type EnumJson = SerdeJson.EnumJson
export type FacesJson = SerdeJson.FacesJson
export type Float32Json = SerdeJson.Float32Json
export type Float64Json = SerdeJson.Float64Json
export type FontJson = SerdeJson.FontJson
export type Int32Json = SerdeJson.Int32Json
export type Int64Json = SerdeJson.Int64Json
export type NumberRangeJson = SerdeJson.NumberRangeJson
export type NumberSequenceJson = SerdeJson.NumberSequenceJson
export type PhysicalPropertiesJson = SerdeJson.PhysicalPropertiesJson
export type RayJson = SerdeJson.RayJson
export type RectJson = SerdeJson.RectJson
export type StringJson = SerdeJson.StringJson
export type UDimJson = SerdeJson.UDimJson
export type UDim2Json = SerdeJson.UDim2Json
export type Vector2Json = SerdeJson.Vector2Json
export type Vector3Json = SerdeJson.Vector3Json
export type Region3Json = SerdeJson.Region3Json
export type InfJson = SerdeJson.InfJson
export type NaNJson = SerdeJson.NaNJson
export type OptionJson<V> = SerdeJson.OptionJson<V>
export type ResultJson<V, E> = SerdeJson.ResultJson<V, E>
export type AnyJson = SerdeJson.AnyJson
export type HeaderMap = Reqwest.HeaderMap
export type Body = Reqwest.Body
export type Json = Reqwest.Json
export type Request = Reqwest.Request
export type RequestBuilder = Reqwest.RequestBuilder
export type Method = Reqwest.Method
export type Response = Reqwest.Response
export type Client = Reqwest.Client
export type ClientBuilder = Reqwest.ClientBuilder
export type Url = Reqwest.Url
export type TryFromUrl = Reqwest.TryFromUrl

return table.freeze({
	Option = Option,
	Result = Result,
	Future = Future,
	Error = Error,
	Vec = Vec,
	HashMap = HashMap,
	VecDeque = VecDeque,
	SerdeJson = SerdeJson,
	Reqwest = Reqwest,
})
